cs3api4lab/auth/reva_password.py

```diff
@@ -0,0 +1,40 @@
+import cs3.gateway.v1beta1.gateway_api_pb2 as cs3gw
+import cs3.gateway.v1beta1.gateway_api_pb2_grpc as cs3gw_grpc
+import cs3.rpc.code_pb2 as cs3code
+import time
+from tornado import web
+
+from cs3api4lab.auth.authenticator import Authenticator
+from cs3api4lab.auth.channel_connector import ChannelConnector
+
+
+class RevaPassword(Authenticator):
```

Comment:
Since the EOS/oAuth token also needs to be converted in a REVA token (by calling the `Authenticate`, as I mention in my review), we should re-use this code. So I would put it in the parent.

Reply:
Is the EOS token equals to Riva token?
If not this is topic to discuss on the weekly meeting because Riva token is similar oAuth with extra information (below is an example of the decoded token). In my opinion, If we need to change the EOS token to Reva token, I can't do it this from the plugin side, (except if EOS token has a username and secret) only from Riva side. Maybe we need a new auth provider. This auth provider gets the EOS token and returns Riva auth token.

`
{
  "aud": "reva",
  "exp": 1602145654,
  "iat": 1602059254,
  "iss": "cernbox.cern.ch",
  "user": {
    "id": {
      "idp": "cernbox.cern.ch",
      "opaque_id": "4c510ada-c86b-4815-8820-......."
    },
    "username": "einstein",
    "mail": "einstein@......",
    "display_name": "Albert Einstein",
    "groups": [
      "sailing-lovers",
      "violin-haters",
      "physics-lovers"
    ]
  }
}
`

Reply:
Technically there is the functionality to get EOS token, but I never talked about that. For this context, there are no EOS tokens. EOS uses oAuth... The EOS file has extra info, but the token itself is pure oAuth, as retrieved from the idp.

But afaik, you always need to exchange oAuth tokens with the REVA ones. That is why I'm saying you need to authenticate as bearer. If you use the revacmd, you will see that you can login with basic or bearer and, if you choose the bearer, you can give the oAuth token as password.

Reply:
I added the oAuth auth class.